clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -362,6 +362,9 @@ else if (strategy == OffsetResetStrategy.LATEST)
 
     public Map<TopicPartition, OffsetAndTimestamp> getOffsetsByTimes(Map<TopicPartition, Long> timestampsToSearch,
                                                                      long timeout) {
+        if (timestampsToSearch.isEmpty())
+            return Collections.EMPTY_MAP;
```

Comment:
`Collections.emptyMap()` is the right one to use. `EMPTY_MAP` should give you a generic warning.